resources/js/Pages/Dashboard.vue

```diff
@@ -107,19 +121,61 @@ import { ref } from 'vue';
                     post_id: id,
                     comment_text: comment,
                 })
-            },
-            // fetch all comments
-            getAllComments() {
-                axios
-                .get('/comments')
-                .then(response => (this.allcomments = response.data))
+                .then(response => (window.location.reload()))
             },
             //hide & show comments section
             showComments(event) {
                 let targetElement = event.target;
                 let nextElement = targetElement.nextElementSibling;
                 nextElement.classList.toggle("show");
+            },
+            //like & dislike functionality
+            submitLikes(id, type) {
+                this.loggedInUser = document.getElementById('user').innerText;
+                //To check duplicate entry in likes table
+                    for(let i=0; i<this.posts.length; i++){
+                        if(type==='Post') {
+                            for(let j=0; j<this.posts[i].likes.length; j++) {
+                                if(this.posts[i].likes[j].user_id == this.loggedInUser && this.posts[i].likes[j].likeable_id == id) {
+                                    this.entryPresent = true;
+                                }
+                            }
+                        } else {
+                            for(let x=0; x<this.posts[i].comments.length; x++) {
+                                if(this.posts[i].comments[x].likes) {
+                                    for(let y=0; y<this.posts[i].comments[x].likes.length; y++) {
+                                        if(this.posts[i].comments[x].likes[y].user_id == this.loggedInUser && this.posts[i].comments[x].likes[y].likeable_id == id) {
+                                            this.entryPresent = true;
+                                        }
+                                    }
+                                }      
+                            }
+                        }
+                }
+                if(!this.entryPresent) {
+                    axios
+                    .post('/submit-likes', {
+                        id: id,
+                        type: type
+                    })
+                    .then(response => (window.location.reload()))
+                }else {
+                    axios
+                    .post('/delete-like',{
+                        id: id
+                    })
+                    .then(response => (window.location.reload()))
+                }
```

Comment:
What I would do here is combine the like and unlike functionality into a single method that sends a request to the backend, then on the backend I would check for the existence of a like in the DB - if there is one present: delete it, otherwise add it.